ladybug/_datacollectionbase.py

```diff
@@ -407,6 +414,74 @@ def are_collections_aligned(data_collections, raise_exception=True):
+        """Compute a function with a list of aligned data collections or individual values.
+
+        Args:
+            funct: A function with a single numerical value as output and one or
+                more numerical values as input.
+            data_collections: A list with a length equal to the number of arguments
+                for the function. Items of the list can be either Data Collections
+                or individual values to be used at each datetime of other collections.
+            result_type: An instance of a Ladybug data type that describes the resulting
+                numerical values.
+            result_unit: The units of the resulting numerical values.
+
+        Return:
+            A Data Collection with the results function. If all items in this list of
+            data_collections are individual values, only a single value will be returned.
+
+        Usage:
+            from ladybug.datacollection import HourlyContinuousCollection
+            from ladybug.epw import EPW
+            from ladybug.psychrometrics import humid_ratio_from_db_rh
+            from ladybug.datatype.percentage import HumidityRatio
+
+            epw_file_path = './epws/denver.epw'
+            denver_epw = EPW(epw_file_path)
+            pressure_at_denver = 85000
+            hr_inputs = [denver_epw.dry_bulb_temperature,
+                         denver_epw.relative_humidity,
+                         pressure_at_denver]
+            humid_ratio = HourlyContinuousCollection.compute_function_aligned(
+                humid_ratio_from_db_rh, hr_inputs, HumidityRatio(), 'fraction')
+            # humid_ratio will be a Data Colleciton of humidity ratios at Denver
+        """
+        # first check the input collections or individual values
+        data_c = []
+        numerical_vals = []
+        for func_input in data_collections:
+            if isinstance(func_input, BaseCollection):
+                data_c.append(func_input)
+            else:
+                try:
+                    numerical_vals.append(float(func_input))
+                except ValueError:
+                    raise TypeError('Expected a number or a Data Colleciton. '
+                                    'Got {}'.format(type(func_input)))
+
+        # run the function and return the result
+        if len(data_c) == 0:
+            # just inidivdual numerical values
+            return funct(*numerical_vals)
+        else:
+            # data collections
+            aligned_collections = []
+            for func_input in data_collections:
+                if isinstance(func_input, BaseCollection):
+                    aligned_collections.append(func_input)
+                else:
+                    aligned_collections.append(
+                        data_c[0].get_aligned_collection(float(func_input)))
+            BaseCollection.are_collections_aligned(aligned_collections)
```

Comment:
Do we really need this double check? since you just created the new `aligned_collections` it better be aligned! If there is a chance that they will end up not being aligned then we should get tit fixed when the issue happens.

Reply:
You are right that this was not the right place for the check.  However, it still needs to be there in case someone connects a list of two or more data collections that are not aligned with one another (otherwise, people will just get an IndexError instead of the more helpful one about non-aligned collections and which collections are problematic).  I have updated it so that it only runs this check for the data collections that are actually input to the method and not collections that are generated from individual float values.